service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -87,6 +87,16 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 	}
 
+	for name, currentEntryCR := range currentEntryCRs {
```

Comment:
Another loop from currentEntryCRs for deletion check.

Reply:
We need to delete to keep the history to 5 but could you add a comment explaining that?

As otherwise it might not be clear why we're deleting when the event is create or update.

Reply:
Sure, I will add a comment below.